### PR TITLE
adds functionality to converter between resources using Json representation

### LINKED ISSUE #556: Feat : Add functionality to convert Hapi resources to Fhir protos (JSON)
**Describe the solution you'd like**

Add functionality to convert Hapi resources to Fhir protos by converting first to JSON and then to then parsing it to the other format. 

**Describe alternatives you've considered**

Use reflection to covert between the resources directly. (will be done later)

FYI @Rkareko

### PR DESCRIPTION
**Pull request by epicadk**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #556 

**Description**
Clear and concise code change description. 

Adds functionality to convert between Fhir protos and Hapi resources

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

Discussed in issue

**Type**
 Feature

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

File: hapiprotoconverter/src/main/java/com/google/android/fhir/hapiprotoconverter/JsonConverter.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+package com.google.android.fhir.hapiprotoconverter
+
+import ca.uhn.fhir.parser.IParser
+import com.google.fhir.common.JsonFormat
+import com.google.fhir.shaded.protobuf.GeneratedMessageV3
+import org.hl7.fhir.r4.model.Resource
+
+@Suppress("UNCHECKED_CAST")
+/**
+ * @param resource resource that will be converter to FhirProto representation
```
Comment (maanuanubhav999) [inline hapiprotoconverter/src/main/java/com/google/android/fhir/hapiprotoconverter/JsonConverter.kt:26]: Converted? Maybe

File: hapiprotoconverter/src/main/java/com/google/android/fhir/hapiprotoconverter/JsonConverter.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+): T {
+  // Generates a new Builder of proto class T
+  val newBuilder =
+    T::class.java.getDeclaredMethod("newBuilder").invoke(null) as GeneratedMessageV3.Builder<*>
+  protoParser.merge(hapiParser.encodeResourceToString(resource), newBuilder)
+  return newBuilder.build() as T
+}
+
+/**
+ * @param resource resource that will be converter to Hapi resource representation
```
Comment (maanuanubhav999) [inline hapiprotoconverter/src/main/java/com/google/android/fhir/hapiprotoconverter/JsonConverter.kt:44]: Same

File: hapiprotoconverter/src/main/java/com/google/android/fhir/hapiprotoconverter/JsonConverter.kt (hunk 1 above)
Comment (epicadk) [reply hapiprotoconverter/src/main/java/com/google/android/fhir/hapiprotoconverter/JsonConverter.kt:26]: Oops sorry thanks!

**Comment by Tarun-Bhardwaj** [comment]
@maanuanubhav999 , @epicadk , is this a GSOC issue pending mentor review? CC @f-odhiambo

**Comment by epicadk** [comment]
> @maanuanubhav999 , @epicadk , is this a GSOC issue pending mentor review? CC @f-odhiambo

Yep, cc @Rkareko.

**Pull request closed without merging**